lib/iris/util.py

```diff
@@ -1554,3 +1554,111 @@ def _meshgrid(*xi, **kwargs):
+        discontiguities in the coordinate points array.
+
+        This can be used as the input array for
+        :func:`iris.util.mask_discontiguities`.
+
+    Examples::
+
+        # Find any unknown discontiguities in your cube's x and y arrays:
+        discontiguities = iris.util.find_discontiguities(cube)
+
+        # Pass the resultant boolean array to `iris.util.mask_discontiguities`
+        # with a cube slice; this will use the boolean array to mask
+        # any discontiguous data points before plotting:
+        masked_cube_slice = iris.util.mask_discontiguities(cube[0],
+                                                           discontiguities)
+
+        # Plot the masked cube slice:
+        iplt.pcolormesh(masked_cube_slice)
+
+    """
+    lats_and_lons = ['latitude', 'grid_latitude',
+                     'longitude', 'grid_longitude']
+    spatial_coords = [coord for coord in cube.aux_coords if
+                      coord.name() in lats_and_lons]
+    dim_err_msg = 'Discontiguity searches are currently only supported for ' \
+                  '2-dimensional coordinates.'
+    if len(spatial_coords) != 2:
+        raise NotImplementedError(dim_err_msg)
+
+    # Check which dimensions are spanned by each coordinate.
+    for coord in spatial_coords:
+        if coord.ndim != 2:
+            raise NotImplementedError(dim_err_msg)
+        else:
+            span = set(cube.coord_dims(coord))
+        if not span:
+            msg = 'The coordinate {!r} doesn\'t span a data dimension.'
+            raise ValueError(msg.format(coord.name()))
+
+    # Check that the 2d coordinate arrays are the same shape as each other
+    if len(spatial_coords) == 2:
+        assert spatial_coords[0].points.shape == spatial_coords[1].points.shape
+
+    # Set up unmasked boolean array the same size as the coord points array:
+    bad_points_boolean = np.zeros(spatial_coords[0].points.shape, dtype=bool)
+
+    for coord in spatial_coords:
+        _, (diffs_x, diffs_y) = coord._discontiguity_in_bounds(rtol=rel_tol,
+                                                               atol=abs_tol)
+
+        bad_points_boolean[:, :-1] = np.logical_or(bad_points_boolean[:, :-1],
+                                                   diffs_x)
+        # apply mask for y-direction discontiguities:
+        bad_points_boolean[:-1, :] = np.logical_or(bad_points_boolean[:-1, :],
+                                                   diffs_y)
+    return bad_points_boolean
+
+
+def mask_cube(cube, points_to_mask):
+    """
+    Masks any cells in the data array which correspond to cells marked `True`
+    in the `points_to_mask` array.
+
+    Args:
+
+    * cube (`iris.cube.Cube`):
+        A 2-dimensional instance of :class:`iris.cube.Cube`.
+
+    * points_to_mask (`numpy.ndarray` of bool):
+        A 2d boolean array of Truth values representing points to mask in the
+        x and y arrays of the cube.
```

Comment:
Surely there's no reason the cube and mask *have* to be 2D. Could you not just say that the mask must match the cube's shape?

Reply:
Actually it's disappointing how *hard* the broadcasting was to get right.
The obvious (to me!) thing, which would look a bit like 
`full_dim_slices = xy_slices[None, None, :, :]; cube[full_dim_slices] = ma.masked`, 
*does not work*.
So I think you need to do an "|=" to get it to broadcast, or something like that, but that means addressing the mask array directly, with all the usual problems that causes.  
So for the sake of speed, as there were still a whole bunch of tiresome issues + choices to be agreed, we gave it up instead :grin: .